### DISCUSSION TITLE
Can I have a regex rule with grouping?

### DISCUSSION THREAD

**Discussion by wsw70** [firefly-iii/firefly-iii#9158] [Firefly III]
My bank provides limited metadata about payment. In particular, the destination is not part of it. It is possible, however, to reasonably match the destination

![image](https://github.com/user-attachments/assets/96c38b9a-b02e-4548-b156-c99f4e24b27f)

In the example above, the message is `CARTE 08/05 AMAZON PAYMENTS PARIS` from which I could extract the destination (`AMAZON PAYMENTS PARIS`) via a regex such as  `CARTE\s\d\d\/\d\d\s(.+)`.

My questions:
- is there a way in an automation to match the description of an operation via a regex
- and can the extracted group be used in the action?

If it is impossible to do it a posteriori, could it be done at ingestion time? (by placing extra metadata which would then be used to set the destination) - a similar question was raised by @shawly in https://github.com/orgs/firefly-iii/discussions/8468 but there was no followup

Ultimately - is there documentation on the data storage itself, to directly add information to metadata via an external program?

**Comment by JC5** [comment]
When you import CSV files, you can do pre-processing on the files yourself. If you use GoCardless or Spectre, you can use webhooks to respond to the created transaction, and submit changes to the API to your liking.

https://docs.firefly-iii.org/how-to/firefly-iii/features/webhooks/
